src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -436,5 +434,14 @@ internal void VerifyWrite(int originalDepth, Utf8JsonWriter writer)
         /// <param name="value">The value to convert.</param>
         /// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
         public abstract void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options);
+
+        internal virtual T ReadWithQuotes(ref Utf8JsonReader reader)
+            => throw new InvalidOperationException();
+
+        internal virtual void WriteWithQuotes(Utf8JsonWriter writer, [DisallowNull] T value, JsonSerializerOptions options, ref WriteStack state)
+            => throw new InvalidOperationException();
+
+        internal sealed override void WriteWithQuotesAsObject(Utf8JsonWriter writer, object value, JsonSerializerOptions options, ref WriteStack state)
+            => WriteWithQuotes(writer, (T)value, options, ref state);
```

Comment:
Can we call these `Read/WriteAsPropertyName()`? I expect the methods to help with the number-as-string feature (https://github.com/dotnet/runtime/issues/30255) to be called `Read/WriteValueWithQuotes`

Reply:
Why not just re-use `ReadWithQuotes`? 
What's the impact of #30255 on serialization? I thought that it would only affect deserialization.

Reply:
I see that `ReadWithQuotes` works since the implementations are agnostic wrt to whether we are looking at a property name vs a property value.

For https://github.com/dotnet/runtime/issues/30255, some folks want to read strings as numbers. This suggests to me that some folks may want to write numbers as strings (which I think is reasonable if we allow the reverse). We should design for this case and include it in the API proposal.

Reply:
We can update the name if needed in a follow up.